libgadget/treewalk.c

```diff
@@ -903,14 +903,11 @@ int treewalk_visit_ngbiter(TreeWalkQueryBase * I,
             /* Skip garbage*/
             if(P[other].IsGarbage)
                 continue;
-
-            /* must be the correct type */
-            if(!((1<<P[other].Type) & iter->mask))
-                continue;
-
-            /* must be the correct time bin */
-            if(lv->tw->type == TREEWALK_SPLIT && !(BINMASK(P[other].TimeBin) & lv->tw->bgmask))
```

Comment:
Are we no longer using TREEWALK_SPLIT?

Reply:
We er never actually used TREEWALK_SPLIT: we never finished implementing it.